### PR TITLE
Added yamllint config file .yamllint, fixed current linting errors, added install and run yamllint targets.

### PR DESCRIPTION
**Pull request by yuanning6**
**What this PR does / why we need it**:

1. This PR adds a .yamllint config file which has rules to config yamllint to enhance linting for yaml files. The lint will be helpful to avoid issues like https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/commit/7cd7fcc1c0448df9caae393fa63720a874b36c85.

2. Fixed current yamllint errors.

3. Added install and run yamllint targerts in Makefile.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/1497

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by yuanning6** [comment]
Hi @Prajyot-Parab !

Thank you so much for your time to take a look at my code! Please let me know if any adjustment is needed!

**Comment by Prajyot-Parab** [comment]
/cc. @Karthik-K-N @Amulyam24

File: Makefile [hunk 1]
```diff
@@ -554,6 +554,24 @@ else
 	echo "Versions are different across Makefiles. Please ensure to keep them uniform."
 endif
 
+
+.PHONY: install-yamllint
+install-yamllint: ## Install yamllint if not present
+	@which yamllint > /dev/null || (echo "Installing yamllint..." && pip install --user yamllint)
```
Comment (Karthik-K-N) [inline Makefile:560]: whats the guarantee that the pip will be available,  aren't there any alternative to download this other than pip install, may be via go?

**Review by Karthik-K-N — COMMENTED** [review]
Thank you for the PR, A quick query other than that overall LGTM. 

Also I see you have previous PR: https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1521, Its advised to update the same PR with latest change but I could see you are hitting some issue and I hope you tried to resolve it before raising this.

If not required close the no more needed PR out of these 2.

**Comment by mkumatag** [comment]
Don't see any activities in this PR, @Amulyam24 will pick this PR and review comments and create a new one

**Pull request closed without merging**